### ISSUE TITLE
Document shrinkwrap best practices

### ISSUE THREAD

**Issue by lkptrzk** [hapijs/contrib#26]
It's not too clear from the [history of npm-shrinkwrap.json in hapijs/hapi](https://github.com/hapijs/hapi/commits/master/npm-shrinkwrap.json) what the guidelines are for maintaining it. Seems like it just snuck in unceremoniously via hapijs/hapi#2039.

I've noticed a few shrinkwrap-related issues filed in the hapi repo due to `npm whatever --production`, so it seems like a relevant thing to document.

Some notes I think would be cool to see:

* tl;dr of why it's used
* When it's okay to change it
* Warning of what kind of bugs it could cause
* Rationale for why it differs from the default `npm shrinkwrap` output

**Comment by hueniverse** [comment]
It's used because it is the only way I have to ensure the version you `npm install` has been fully verified by me. Otherwise, with so many maintainers and dependencies, you can never know if your integrated hapi experience is valid and secure.

You can change or ignore it if you know what you are doing (e.g. tested and verified the changes not included in it yourself).

The format is manually modified to allow easier maintenance.

**Comment by lkptrzk** [comment]
thanks
